### ISSUE TITLE
Unnessecary table.unpack when passing `...` parameter to a function local variable

### ISSUE THREAD

**Issue by GlassBricks** [TypeScriptToLua/TypeScriptToLua#1047]
See here:
https://typescripttolua.github.io/play/#code/FAMwrgdgxgLglgewgAgG4EMBOWDmAVdAazghwAoA6K3AZwC5l0IBPAbQF0BKZAbwF9goSLEQoQCBAEZK1TDnqMWHbj2DI0WXAWKkZFWp2ACh0eEmTiEAJj20GTNl15rkUJDRgXkAXg3Y52iQ4jDTItnIKDso+AHxoCHAAJi4g4fKGAkA

When `varargTaking` is called directly, the ... is passed directly. However, when using a local variable with function type instead, a unnessecary `table.unpack` is generated.

**Comment by Perryvw** [comment]
Hopefully this will be solved by #1046 that I still have to look at, me or @tomblind can have a look at verifying this example on that branch too.

**Comment by tomblind** [comment]
Yes, this is fixed by that PR. This is actually the other case I found where the function type could be defined inside the outer function and trigger de-optimization.

Here's the output when the fix is applied:
```lua
function varargTaking(self, ...)
end
function foo1(self, ...)
    varargTaking(_G, ...)
end
function foo2(self, ...)
    local f = varargTaking
    f(_G, ...)
end
```
